app/src/test/java/org/mozilla/focus/toolbar/BrowserAppBarViewModelTest.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.toolbar
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class BrowserAppBarViewModelTest {
+
+    @get:Rule val instantTaskRule = InstantTaskExecutorRule() // necessary for LiveData tests.
+
+    private lateinit var viewModel: BrowserAppBarViewModel
+
+    @Before
+    fun setUp() {
+        viewModel = BrowserAppBarViewModel()
+    }
+
+    @Test
+    fun `GIVEN voiceView is disabled WHEN navigation overlay is visible THEN toolbar scroll is disabled`() {
+        viewModel.setIsVoiceViewEnabled(false)
+        viewModel.setIsNavigationOverlayVisible(true)
+        viewModel.isToolbarScrollEnabled.observeForever { assertFalse(it!!) }
+    }
+
+    @Test
+    fun `GIVEN voiceView is disabled WHEN navigation overlay is not visible THEN toolbar scroll is enabled`() {
+        viewModel.setIsVoiceViewEnabled(false)
+        viewModel.setIsNavigationOverlayVisible(false)
+        viewModel.isToolbarScrollEnabled.observeForever { assertTrue(it!!) }
+    }
+
+    @Test
+    fun `GIVEN voiceView is enabled THEN toolbar scroll is always disabled`() {
+        viewModel.setIsVoiceViewEnabled(true)
+        viewModel.setIsNavigationOverlayVisible(false)
+        viewModel.isToolbarScrollEnabled.observeForever { assertFalse(it!!) }
+        viewModel.setIsNavigationOverlayVisible(true)
```

Comment:
It took me a moment to realize that the observeForever applied to all states here, and in more complex tests this can get pretty crazy.  You might want to consider poaching org.mozilla.tv.firefox.helpers.ext.LiveDataTestHelper, where the equivalent test would look like:

```
viewModel.isToolbarScrollEnabled.assertValues(false, false, false, false) {
        viewModel.setIsVoiceViewEnabled(true)
        viewModel.setIsNavigationOverlayVisible(false)
        viewModel.setIsNavigationOverlayVisible(true)
}
```

(The first false is the default state, and the test would fail if more or fewer emissions than expected were encountered)

Reply:
I filed https://github.com/mozilla-mobile/android-components/issues/1725 to put a solution in components. For now, I'll copy this over, thanks.

Reply:
Nope, I think this is fine for now.  I think a more readable, long term solution would be adding a test helper variant that explicitly ignores the default value, but I don't see it as a high priority.